cmd/antrea-agent/options_test.go

```diff
@@ -343,3 +348,56 @@ func TestOptionsValidateSecondaryNetworkConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestOptionsUnmarshalling(t *testing.T) {
+	t.Run("protocolFilter can be unmarshalled", func(t *testing.T) {
+		configYML, err := generateAntreaConf()
+		if err != nil {
+			t.Error("Failed to generate config from build yaml")
+		}
+		agentConfig := agentconfig.AgentConfig{}
+		err = yaml.Unmarshal([]byte(configYML), &agentConfig)
+		if err != nil {
+			t.Error("failed to unmarshal config file")
+		}
+		assert.Contains(t, agentConfig.FlowExporter.ProtocolFilter, "tcp")
+	})
+}
+
+func generateAntreaConf() (string, error) {
+	generatedAntreaConfYML := "../../build/yamls/antrea.yml"
+	startString := "antrea-agent.conf"
+	endString := "antrea-cni.conflist"
+	appendToString := "protocolFilter:"
```

Comment:
unit tests such as the ones in this file should really not have a dependency on an unrelated file like antrea.yml